src/types/typeobject.rs

```diff
@@ -39,8 +37,11 @@ impl PyType {
     }
 
     /// Gets the name of the `PyType`.
-    pub fn name(&self) -> Cow<str> {
-        unsafe { CStr::from_ptr((*self.as_type_ptr()).tp_name).to_string_lossy() }
+    pub fn name(&self) -> &str {
+        self.getattr("__qualname__")
+            .expect("Error accessing __qualname__")
+            .extract()
+            .expect("__qualname__ not a string")
```

Comment:
How about returning `PyResult<&str>`?
It's not fun to see the program crashed in such a trivial operation.
```suggestion
    pub fn name(&self) -> PyResult<&str> {
        self.getattr("__qualname__")?.extract()
```